services/wfm/src/test/java/org/openkilda/wfm/topology/flowhs/utils/RerouteRetryQueueTest.java

```diff
@@ -0,0 +1,167 @@
+
+        reroute = queue.remove();
+        Assert.assertTrue(reroute.isPresent());
+        Assert.assertSame(reroutePathA, reroute.get());
+        Assert.assertFalse(queue.remove().isPresent());
+        Assert.assertEquals(0, queue.size());
+
+        Assert.assertFalse(queue.remove().isPresent());
+
+        // more than 2 entry
+        queue.add(rerouteEmpty);
+        queue.add(reroutePathA);
+        queue.add(reroutePathB);
+
+        reroute = queue.remove();
+        Assert.assertTrue(reroute.isPresent());
+        Assert.assertSame(rerouteEmpty, reroute.get());
+        Assert.assertEquals(1, queue.size());
+
+        reroute = queue.remove();
+        Assert.assertTrue(reroute.isPresent());
+        Assert.assertNotEquals(reroutePathA, reroute.get());  // merged entry
+        Assert.assertNotEquals(reroutePathB, reroute.get());  // merged entry
+        Assert.assertFalse(queue.remove().isPresent());
+        Assert.assertEquals(0, queue.size());
+
+        Assert.assertFalse(queue.remove().isPresent());
+    }
+
+    @Test
+    public void getOperations() {
+        RerouteRetryQueue queue = new RerouteRetryQueue();
+        Assert.assertFalse(queue.get().isPresent());
+
+        Optional<FlowRerouteFact> reroute;
+        queue.add(rerouteEmpty);
+        reroute = queue.get();
+        Assert.assertTrue(reroute.isPresent());
+        Assert.assertSame(rerouteEmpty, reroute.get());
+
+        queue.add(reroutePathA);
+        reroute = queue.get();
+        Assert.assertTrue(reroute.isPresent());
+        Assert.assertSame(rerouteEmpty, reroute.get());
+
+        queue.remove();
+        reroute = queue.get();
+        Assert.assertTrue(reroute.isPresent());
+        Assert.assertSame(reroutePathA, reroute.get());
+    }
+
+    @Test
+    public void mergeOperation() {
+        RerouteRetryQueue queue = new RerouteRetryQueue();
+        queue.add(rerouteEmpty);
+        queue.add(reroutePathA);
+        queue.add(rerouteForced);
+        queue.add(reroutePathB);
+
+        Optional<FlowRerouteFact> potential;
+        queue.remove();
+        potential = queue.get();
+        Assert.assertTrue(potential.isPresent());
+
+        FlowRerouteFact reroute = potential.get();
+        Assert.assertEquals(reroutePathB.getKey(), reroute.getKey());
+        Assert.assertSame(reroutePathB.getCommandContext(), reroute.getCommandContext());
+        Assert.assertEquals(reroutePathB.getFlowId(), reroute.getFlowId());
+
+        Set<PathId> expectedPaths = Stream.of(rerouteEmpty, reroutePathA, rerouteForced, reroutePathB)
+                .map(FlowRerouteFact::getPathsToReroute)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+        Assert.assertEquals(expectedPaths, reroute.getPathsToReroute());
+        Assert.assertTrue(reroute.isForceReroute());
+        Assert.assertEquals(reroutePathB.getRerouteReason(), reroute.getRerouteReason());
```

Comment:
all 4 rerouteFact objects has same reason "reason 1". So check if we set right reason you need to have different reasons